providers/ionoscloud/ionoscloud_service.go

```diff
@@ -160,6 +165,20 @@ func NewClientByType(username, password, token, url string, clientType clientTyp
 				"terraformer_ionos-cloud-sdk-go/%s_os/%s_arch/%s", ionoscloud.Version, runtime.GOOS, runtime.GOARCH)
 			return dataPlatform.NewAPIClient(newConfig)
 		}
+	case dnsClient:
+		{
+			newConfig := dns.NewConfiguration(username, password, token, url)
+
+			if os.Getenv(helpers.IonosDebug) != "" {
+				newConfig.Debug = true
+			}
+			newConfig.MaxRetries = helpers.MaxRetries
+			newConfig.WaitTime = helpers.MaxWaitTime
+			newConfig.HTTPClient = &http.Client{Transport: CreateTransport()}
+			newConfig.UserAgent = fmt.Sprintf(
+				"terraformer_ionos-cloud-sdk-go/%s_os/%s_arch/%s", ionoscloud.Version, runtime.GOOS, runtime.GOARCH)
```

Comment:
dns

Reply:
@cristiGuranIonos solved in cfae8bca1f75b16a9a91814ca20c8513b7bce72b